index.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const assert = require('assert');
+
+class DejaVu {
+
+  /**
+   * Constructs an new instance of DejaVu to identify "original" events.
+   * @param {Object} options The constructor options.
+   * @param {Object} options.redisConnection A connection to a Redis deployment.
+   */
+  constructor(options) {
+    assert(options.redisConnection, 'Must provide a Redis connection');
+    this._redisConnection = options.redisConnection;
+    this._handlers = {};
+  }
+
+  /**
+   * isNew returns if an event is new or not.
+   * @param {String} prefix The prefix to namespace events under.
+   * @param {Number} window The window to filtet event timestamps by (in ms).
```

Comment:
If you checkout the the handler registration function, we handle setting the `ttl` for users if they do not provide a window. This is also provided as these use separate units, and as this module is generalizing event checking, I figured it would be at least nice to give users the option. I also don't envision users using this function directly, it's simply provided to give users access to the lower level primitives. I'll add a README showing desired usage.